src/sagas/CreateClaim.saga.test.ts

```diff
@@ -0,0 +1,179 @@
+import { Frost } from '@po.et/frost-client'
+import { delay } from 'redux-saga'
+import { takeLatest, call, put } from 'redux-saga/effects'
+import { describe } from 'riteway'
+const { toast } = require('react-toastify')
+
+import { Actions } from '../actions/index'
+import { CreateClaimSaga, handleOnCreateClaim, handleOnCreateClaimSuccess, handleOnCreateClaimError } from './CreateClaim.saga'
+import { WorkAttributes } from '../interfaces/Props'
+import { Configuration } from '../configuration'
+
+const createWork = ({
+  name = 'name',
+  dateCreated = new Date().toISOString(),
+  datePublished = new Date().toISOString(),
+  content = 'test',
+  tags = 'test',
+  author = 'author'
+}: WorkAttributes = {
+  name: 'name',
+  dateCreated: new Date().toISOString(),
+  datePublished: new Date().toISOString(),
+  content: 'test',
+  tags: 'test',
+  author: 'author'
+}) => ({
+  name,
+  dateCreated,
+  datePublished,
+  content,
+  tags,
+  author
+})
+
+describe('CreateClaimSaga Saga', async assert => {
+  const iterator = CreateClaimSaga()();
+  
+  assert({
+    given: 'Work Claim Form Submit Action',
+    should: 'handle post work request',
+    actual: iterator.next().value,
+    expected: takeLatest(Actions.CreateClaim.CREATE_CLAIM, handleOnCreateClaim),
+  })
+
+  assert({
+    given: 'Work Claim Form Submit Success Action',
+    should: 'handle post work request',
+    actual: iterator.next().value,
+    expected: takeLatest(Actions.CreateClaim.CREATE_CLAIM_SUCCESS, handleOnCreateClaimSuccess),
+  })
+
+  assert({
+    given: 'Work Claim Form Submit Error Action',
+    should: 'handle post work request',
+    actual: iterator.next().value,
+    expected: takeLatest(Actions.CreateClaim.CREATE_CLAIM_ERROR, handleOnCreateClaimError),
+  })
+})
+
+describe('handleOnCreateClaim()', async assert => {
+  const token = 'test-token'
+  const work = createWork()
+  const workId = 'testWorkId'
+
+  const iterator = handleOnCreateClaim(Actions.CreateClaim.onCreateClaim({ work, token }))
+
+  assert({
+    given: 'Work Claim Form Submit Action',
+    should: 'set loading status',
+    actual: iterator.next().value,
+    expected: put(Actions.LoadingPage.onLoadingOn()),
+  })
+
+  {
+    const frost = new Frost({ host: Configuration.frostApiUrl })
```

Comment:
can this be moved out of the describe, to the top of the file? That way it can be used for other tests if needed later on.

Reply:
if there is a reason you placed it here that im not seeing let me know.